backend/opampd/protocol.go

```diff
@@ -56,3 +67,31 @@ func (p *Protocol) OnAgentDisconnect(instanceUid string, disconnect *protobufs.A
 	//TODO implement me
 	return nil, fmt.Errorf("implement me")
 }
+
+func (p *Protocol) createEntity(instanceUid string, report *protobufs.StatusReport) error {
+	entity := &corev2.Entity{
+		EntityClass:        corev2.EntityAgentClass,
+		System:             corev2.System{},
+		Subscriptions:      []string{"opamp"},
+		LastSeen:           0,
+		Deregister:         true,
+		Deregistration:     corev2.Deregistration{},
+		User:               "",
+		ExtendedAttributes: nil,
+		Redact:             nil,
+		ObjectMeta: corev2.ObjectMeta{
+			Name:      instanceUid,
+			Namespace: entityNamespace,
+			Labels: map[string]string{
+				"opamp-instanceid": instanceUid,
+				"opamp-agent-type": report.AgentDescription.AgentType,
+			},
+			Annotations: nil,
+			CreatedBy:   "",
+		},
+		SensuAgentVersion: report.AgentDescription.AgentVersion,
```

Comment:
We shouldn't reuse this class and sensu agent specific fields. Something closure to a proxy entity makes more sense.

Reply:
For now is it ok if I add this information as a `opamp-agent-version` label?